src/ORM/Association/SelectableAssociationTrait.php

```diff
@@ -239,8 +239,8 @@ protected function _buildSubquery($query)
             $keys = (array)$this->foreignKey();
         }
 
-        $fields = $query->aliasFields($keys);
-        return $filterQuery->select($fields, true);
+        $fields = $query->aliasFields($keys, $this->source()->alias());
+        return $filterQuery->select($fields, true)->distinct();
```

Comment:
`inner join (select author.id from articles join authors)` will return the same `author.id`, thus duplicating rows. Because of the bug we had, that problem was not obvious.